lib/sles4sap/console_redirection.pm

```diff
@@ -0,0 +1,174 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::console_redirection;
+use strict;
+use warnings;
+use testapi;
+use Carp qw(croak);
+use Exporter qw(import);
+use serial_terminal qw(select_serial_terminal);
+use Regexp::Common qw(net);
+
+=head1 SYNOPSIS
+
+Library that enables console redirection and file transfers from worker based VM to another host.
+Can be used for cases where worker VM is not the target host for API calls and command execution, but serves only as a jumphost.
+
+=cut
+
+our @EXPORT = qw(
+  connect_target_to_serial
+  disconnect_target_from_serial
+  verify_ip
+  redirection_init
+  check_serial_redirection
+);
+
+my $ssh_opt = '-o StrictHostKeyChecking=no -o ServerAliveInterval=60 -o ServerAliveCountMax=120';
+
+=head2 handle_login_prompt
+
+    handle_login_prompt();
+
+B<ssh_user> Login user
+Detects if login prompt appears and types the password.
+
+=cut
+
+sub handle_login_prompt {
+    my $pwd = get_var('_SECRET_SUT_PASSWORD', $testapi::password);
+    set_serial_term_prompt();
+    my $serial_response = wait_serial(qr/Password:\s*$|:~/i);
+    # Handle password prompt if it appears
+    if (grep /Password:\s*$/, $serial_response) {
+        type_password $pwd;
+        send_key 'ret';
+    }
+    croak 'Failed to setup serial console' unless grep /:~/, $serial_response;
+}
+
+=head2 redirection_init
+
+    redirection_init();
+
+Do preparation before redirecting console. Gets base VM id and initial setup.
+Needs to be done only once.
```

Comment:
`Needs to be done only once.` Does means "please only call me once" or "no matter how many time you call me only the first do something"? What "once" refer too? "Once" per scheduling? "Once" per tunnel?

Why `Needs to be done only once.` ? What bad things could happens if it is not only once?

Reply:
I see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18815/files#diff-ec77a3b424eb67f5827020a80f6d82a96f2d51bbff53a2ea1f1979bed039bf14R101

but I still do not fully understand.

Reply:
Once per whole test. The point is to get machine-id of the local worker VM so the `disconnect_target_from_serial` knows when to stop logging out from ssh connections. 

I have rewritten the description.